### ISSUE TITLE
Trivy config scan gets SIGSEGV

### ISSUE THREAD

**Issue by raesene** [aquasecurity/trivy#1113]
## Description

Running the Trivy configuration scanner on a directory of manifests generates a runtime error. Other scans (e.g. Docker image scan) seem to work ok.

```
trivy config manifests/
2021-07-12T13:16:34.636+0100    INFO    Need to update the built-in policies
2021-07-12T13:16:34.636+0100    INFO    Downloading the built-in policies...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1634bfa]

goroutine 1 [running]:
github.com/aquasecurity/trivy/pkg/policy.Client.DownloadBuiltinPolicies(0x0, 0x0, 0x1d39030, 0x2860b98, 0x1d2e838, 0xc00008a660, 0x1, 0x0)
        /home/runner/work/trivy/trivy/pkg/policy/policy.go:161 +0x3a
github.com/aquasecurity/trivy/pkg/commands/operation.InitBuiltinPolicies(0x1d2e838, 0xc00008a660, 0xc000597100, 0x1, 0x1, 0x1, 0x1, 0x0)
        /home/runner/work/trivy/trivy/pkg/commands/operation/operation.go:124 +0x412
github.com/aquasecurity/trivy/pkg/commands/artifact.scan(0x1d2e838, 0xc00008a660, 0xc0005ad780, 0xc0001820b8, 0x1cd4c60, 0x6, 0x0, 0xc0001802a0, 0x12, 0x0, ...)
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:162 +0x80d
github.com/aquasecurity/trivy/pkg/commands/artifact.runWithTimeout(0x1d2e838, 0xc00008a660, 0xc0005ad780, 0xc0001820b8, 0x1cd4c60, 0x6, 0x0, 0xc0001802a0, 0x12, 0x0, ...)
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:67 +0x308
github.com/aquasecurity/trivy/pkg/commands/artifact.Run(0x1d2e800, 0xc00003c0d0, 0xc0005ad780, 0xc0001820b8, 0x1cd4c60, 0x6, 0x0, 0xc0001802a0, 0x12, 0x0, ...)
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:39 +0xf8
github.com/aquasecurity/trivy/pkg/commands/artifact.ConfigRun(0xc0005ad780, 0x13, 0x24)
        /home/runner/work/trivy/trivy/pkg/commands/artifact/config.go:30 +0x2d6
github.com/urfave/cli/v2.(*Command).Run(0xc0005cb0e0, 0xc0005ad040, 0x0, 0x0)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x4dd
github.com/urfave/cli/v2.(*App).RunContext(0xc0002fc340, 0x1d2e800, 0xc00003c0d0, 0xc00003a090, 0x3, 0x3, 0x0, 0x0)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0x810
github.com/urfave/cli/v2.(*App).Run(...)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
        /home/runner/work/trivy/trivy/cmd/trivy/main.go:16 +0x87
```

## What did you expect to happen?

An analysis of the manifest(s) in the directory specified

## What happened instead?

Panic!

## Output of run with `-debug`:

```
trivy -debug config manifests/
2021-07-12T13:19:35.201+0100    DEBUG   Severities: UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL
2021-07-12T13:19:35.213+0100    DEBUG   cache dir:  /root/.cache/trivy
2021-07-12T13:19:35.214+0100    DEBUG   Vulnerability type:  []
2021-07-12T13:19:35.214+0100    DEBUG   Failed to open the policy metadata: open /root/.cache/trivy/policy/metadata.json: no such file or directory
2021-07-12T13:19:35.214+0100    INFO    Need to update the built-in policies
2021-07-12T13:19:35.214+0100    INFO    Downloading the built-in policies...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1634bfa]

goroutine 1 [running]:
github.com/aquasecurity/trivy/pkg/policy.Client.DownloadBuiltinPolicies(0x0, 0x0, 0x1d39030, 0x2860b98, 0x1d2e838, 0xc00063c7e0, 0x1, 0x0)
        /home/runner/work/trivy/trivy/pkg/policy/policy.go:161 +0x3a
github.com/aquasecurity/trivy/pkg/commands/operation.InitBuiltinPolicies(0x1d2e838, 0xc00063c7e0, 0xc00065b500, 0x1, 0x1, 0x1, 0x1, 0x0)
        /home/runner/work/trivy/trivy/pkg/commands/operation/operation.go:124 +0x412
github.com/aquasecurity/trivy/pkg/commands/artifact.scan(0x1d2e838, 0xc00063c7e0, 0xc00003aa40, 0xc0006a41a0, 0x1cd4c60, 0x6, 0x100, 0xc00003e4e0, 0x12, 0x0, ...)
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:162 +0x80d
github.com/aquasecurity/trivy/pkg/commands/artifact.runWithTimeout(0x1d2e838, 0xc00063c7e0, 0xc00003aa40, 0xc0006a41a0, 0x1cd4c60, 0x6, 0x100, 0xc00003e4e0, 0x12, 0x0, ...)
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:67 +0x308
github.com/aquasecurity/trivy/pkg/commands/artifact.Run(0x1d2e800, 0xc00003c0d0, 0xc00003aa40, 0xc0006a41a0, 0x1cd4c60, 0x6, 0x100, 0xc00003e4e0, 0x12, 0x0, ...)
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:39 +0xf8
github.com/aquasecurity/trivy/pkg/commands/artifact.ConfigRun(0xc00003aa40, 0x13, 0x24)
        /home/runner/work/trivy/trivy/pkg/commands/artifact/config.go:30 +0x2d6
github.com/urfave/cli/v2.(*Command).Run(0xc000637200, 0xc00027fe00, 0x0, 0x0)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x4dd
github.com/urfave/cli/v2.(*App).RunContext(0xc00035d6c0, 0x1d2e800, 0xc00003c0d0, 0xc000188040, 0x4, 0x4, 0x0, 0x0)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0x810
github.com/urfave/cli/v2.(*App).Run(...)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
        /home/runner/work/trivy/trivy/cmd/trivy/main.go:16 +0x87
```

## Output of `trivy -v`:

```
trivy -v
Version: 0.19.0
Vulnerability DB:
  Type: Light
  Version: 1
  UpdatedAt: 2021-07-12 12:09:42.936647196 +0000 UTC
  NextUpdate: 2021-07-12 18:09:42.936646996 +0000 UTC
  DownloadedAt: 2021-07-12 12:15:40.0638488 +0000 UTC
```

## Additional details (base image name, container registry info...):

AMD64 version of trivy running in WSL on Windows 10.

**Comment by knqyf263** [comment]
Oh, thanks. I'll have a look.

**Comment by danielhass** [comment]
Ran into the same issue on Arch Linux with the binary release `trivy_0.19.0_Linux-64bit.tar.gz`.

**Comment by miglen** [comment]
Ran into the same issue on macOS with Big Sur 11.3.1 (20E241)
```
 trivy --version
Version: 0.19.0
Vulnerability DB:
  Type: Light
  Version: 1
  UpdatedAt: 2021-07-12 12:09:42.936647196 +0000 UTC
  NextUpdate: 2021-07-12 18:09:42.936646996 +0000 UTC
  DownloadedAt: 2021-07-12 15:44:19.036633 +0000 UTC
 ```

**Comment by knqyf263** [comment]
It was a really stupid mistake. Hopefully, this bug was fixed at v0.19.1. Please check it out.
https://github.com/aquasecurity/trivy/releases/tag/v0.19.1

Thanks for all your help.
